adminstatus/render_adminpage.py

```diff
@@ -57,26 +65,32 @@ def render_adminpage(output_filename):
             print("Improper configuration for user: {}".format(user_name))
             exit_status = 1
             continue
-        user = dict(name=user_name, expire=user_expire, status=user_status)
-        users.append(user)
-        users_to_send.append([user_name, expire_string])
+        finally:
+            user = dict(name=user_name, expire=user_expire, status=user_status)
+            users.append(user)
+            users_to_send.append([user_name, expire_string])
 
     # convert to JSON to be sent
     user_info = json.dumps(users_to_send)
```

Comment:
Unfortunately, calling `json.dumps` emits characters that do not play nicely when written into a HTML document and the complete data is not sent. As a fix, I would suggest Base64 encoding the output from `json.dumps` and placing it in between double-quotes.

Reply:
The data should be sent now, I converted the output from json.dumps to a bytes-like object and then encoded with base64 like you suggested